### PR TITLE
Handle all git change types in Bitbucket PRs

### LINKED ISSUE #3081: No codeintel on deleted lines in Bitbucket PRs
Added and unchanged lines work, but deleted lines throw
```
hover.ts:49 Error: document not found: git://localhost/SOURCEGRAPH/mux?c5c6c98bc25355028a63748a498942a6398ccd22#mux.go
    at ExtDocuments.get (blob:chrome-extension://bmfbcejdknlknpncfpeloejonjoledha/d2eac217-b3f4-4b7e-98f5-d7c10d4d94d3:27398:13)
    at ExtDocuments.<anonymous> (blob:chrome-extension://bmfbcejdknlknpncfpeloejonjoledha/d2eac217-b3f4-4b7e-98f5-d7c10d4d94d3:27421:19)
    at Generator.next (<anonymous>)
    at fulfilled (blob:chrome-extension://bmfbcejdknlknpncfpeloejonjoledha/d2eac217-b3f4-4b7e-98f5-d7c10d4d94d3:23163:24)
```

### LINKED ISSUE #3084: Bitbucket fetchDiffFiles() fetches file contents serially instead of in parallel
This uses `switchMap` to fetch the data serially. It should be using `zip()` or `forkJoin()`.

https://github.com/sourcegraph/sourcegraph/blob/2047b2bcbeeaa3474310b3ba21d4cb88c5aae772/client/browser/src/libs/bitbucket/file_info.ts#L27-L57

### PR DESCRIPTION
**Pull request by felixfbecker**
Codeintel on Bitbucket PRs didn't work for many git change types.

This makes it work for all change types.
It also splits up the code view specs so there's one per view.
There is a unit test per change type for PR code views.
It would be nice to have these for each code view though, and there is more potential for unit-testing the small DOM inspection functions.
Also adds a lot of comments.

Fixes #3081 
Fixes #3084

### CODE REVIEW

File: client/browser/src/libs/bitbucket/scrape.ts [hunk 1]
```diff
@@ -100,87 +106,189 @@ export const getFileInfoFromCodeView = (
...
+ */
+export const getMostFileInfoFromMultiFileDiffCodeView = (
+    codeViewElement: HTMLElement
+): BitbucketRepoInfo & Pick<FileInfo, 'repoName' | 'baseRepoName' | 'filePath' | 'baseFilePath' | 'rev'> => {
+    // Get the file path from the breadcrumbs
+    const breadcrumbsElement = codeViewElement.querySelector('.breadcrumbs')
+    if (!breadcrumbsElement) {
+        throw new Error('Could not find diff code view breadcrumbs element through selector .breadcrumbs')
     }
+    const filePath = breadcrumbsElement.textContent!
```
Comment (lguychard) [inline client/browser/src/libs/bitbucket/scrape.ts]: Consider adding a check here to verify that `filePath` is not empty

File: client/browser/src/libs/bitbucket/scrape.ts [hunk 2]
```diff
@@ -100,87 +106,189 @@ export const getFileInfoFromCodeView = (
...
-        repoSlug = info.repoSlug
-        commitID = info.commitID
+/**
+ * Gets most of the file info from the DOM of a multi-file diff code view (PR, compare or commit page).
+ *
+ * The returned file info does not have the commit ID and base commit ID.
+ * Those need to be fetched from the Bitbucket API for PRs,
+ * or taken from links on the page for compare and commit pages {@link getCommitInfoFromComparePage}.
+ */
+export const getMostFileInfoFromMultiFileDiffCodeView = (
```
Comment (lguychard) [inline client/browser/src/libs/bitbucket/scrape.ts]: Consider making the naming more explicit: `getFileInfoWithoutCommitIDsFromMultiFileDiff()`

File: client/browser/src/libs/bitbucket/scrape.ts [hunk 3]
```diff
@@ -100,87 +106,189 @@ export const getFileInfoFromCodeView = (
...
+const getChangeTypeElement = ({ codeViewElement }: { codeViewElement: HTMLElement }): HTMLElement | null =>
+    codeViewElement.querySelector<HTMLElement>('.change-type-lozenge')
 
-    const commitID = commitMatch ? commitMatch[1] : undefined
+/**
+ * Reads the change type from the change type indicator badge.
+ */
+const getChangeType = ({ changeTypeElement }: { changeTypeElement: HTMLElement | null }): ChangeType => {
+    if (!changeTypeElement) {
+        return 'MODIFY'
```
Comment (lguychard) [inline client/browser/src/libs/bitbucket/scrape.ts]: This implies that there is no change type element when the change type is `'MODIFY'`. Is this the case? Or should this be an error case?

File: client/browser/src/libs/bitbucket/scrape.ts [hunk 4]
```diff
@@ -100,87 +106,189 @@ export const getFileInfoFromCodeView = (
...
+ */
+export const getFileInfoFromSingleFileDiffCodeView = (
+    codeViewElement: HTMLElement
+): BitbucketRepoInfo & Pick<FileInfo, 'repoName' | 'baseRepoName' | 'filePath' | 'baseFilePath' | 'commitID'> => {
+    const { repoName, project, repoSlug, filePath } = getFileInfoFromLinkInSingleFileView(codeViewElement)
+    const commitID = getCommitIDFromLink()
+    const changeTypeElement = getChangeTypeElement({ codeViewElement })
+    const changeType = getChangeType({ changeTypeElement })
+    const baseFilePath = getBaseFilePathForDiffCodeView({ changeTypeElement, changeType, filePath })
+    const baseRepoName = changeType !== 'ADD' ? repoName : undefined
```
Comment (lguychard) [inline client/browser/src/libs/bitbucket/scrape.ts]: Consider explaining in a comment why `baseRepoName` should be `undefined` when `changeType === 'ADD'`

File: client/browser/src/libs/bitbucket/scrape.ts (hunk 3 above)
Comment (felixfbecker) [reply client/browser/src/libs/bitbucket/scrape.ts]: Did you see this comment? https://github.com/sourcegraph/sourcegraph/pull/3086/files/36dcc2e77f9210318ade81838294591c5656ad4f#diff-2ca53c5c095296eeb58541d5a3cca52dR137

Comment (lguychard) [reply client/browser/src/libs/bitbucket/scrape.ts]: Nope, I see, thanks!

**Pull request merged**